### ISSUE TITLE
Mover need to be ensemble specific

### ISSUE THREAD

**Issue by jhprinz** [choderalab/openpathsampling#272]
#### Question:

Would it be okay or desirable if all movers need to be told which ensembles to use explicitly on creation?

#### Example
```
randomrepex = ReplicaExchangeMover([[ens1, ens2] , [ens2, ens3], [ens3, ens4]])
```
would have to be written explicitly
```
randomrepex = RandomChoiceMover([
    ReplicaExhangeMover(ens1, ens2),
    ReplicaExhangeMover(ens2, ens3),
    ReplicaExhangeMover(ens3, ens4)
])
```
so that each mover knows exactly which ensembles it acts upon and creates. Very useful for cleaner implementation using input and output ensembles

**Comment by dwhswenson** [comment]
If I understand, you're asking whether we should say that each instance of a mover is tied to the most atomic level of trial; i.e., the instance can only represent a move for a single, specific ensemble for shooting/path reversal, or a specific pair for replica exchange, or a specific to-from pair for hopping. Is that correct?

I'm very much in favor of this. When we were first implementing this, I thought that others didn't like that approach, so I tried to implement things in such a way that it could work either way (but then made the default setups so that they would use this approach, because it adds a lot of flexibility).

This also simplifies the code significantly. I wrote some fancy stuff to select the correct samples. It was a pain to wrap my head around that code when I wrote it, and I really hoped it was correct, because trying to understand it again to fix it later would be a nightmare.

The only caveat is that, in that sample selection process, you still need to keep in mind the possibility that there is more than one sample in a given ensemble. It should get a list of all active samples in that ensemble, and pick one of them at random. (Think AMS, for example, where you need pick a random trajectory in the ensemble to start your next trajectory from.) But I think that's what SampleSet gives you anyway, isn't it?

**Comment by jhprinz** [comment]
Precisely, that was my idea. And the problem I had is, that the code gets longer and simple things create long lists. On the other hand it is much clearer and will remove the kind of hidden selection of ensemble pairs and put it explicitely into the move scheme. 

As you said, writing a simple to use selection is very diffiult and in the last PR I ran into the same problem (with no satisfying answer). 

The main reason I had to require this is that it make the input output ensemble treatment much simpler.

Concernon your last question. This is no problem since the selection process is now based only on ensembles and given the unique ensemble each mover picks one random allowed sample as we do before and this sample is indexed by its replica number. When creating new trials they will use the replica numbers from the selected samples. So, I am using (also right now) what you implemented in SampleSet and I think that is what we want for using TIS and AMS.

I will make a PR to handle this if you have no more objections. Great...

**Comment by dwhswenson** [comment]
One small point: my MISTIS branch is making some changes that might conflict with this (in particular, I'll be making significant changes to the minus move so that it can support multiple interface set minus.) It might be best to hold off on this until after that gets merged (probably early next week).

**Comment by jhprinz** [comment]
Sure... I will leave this for now, is almost ready and can be finished soon.

**Comment by dwhswenson** [comment]
Solved by #278.
